src/Templates/BlockTemplatesCompatibility.php

```diff
@@ -369,31 +369,42 @@ protected function inject_attribute( &$block ) {
 	 * @return string Wrapped template content inside a div.
 	 */
 	public static function wrap_single_product_template( $template_content ) {
-		$parsed_blocks   = parse_blocks( $template_content );
-		$splitted_blocks = self::split_blocks_by_type( $parsed_blocks );
-
-		$parsed_block_without_template_parts = array_filter(
-			$parsed_blocks,
-			function( $block ) {
-				return 'core/template-part' !== $block['blockName'] && ! empty( $block['blockName'] );
-			}
-		);
+		$parsed_blocks  = parse_blocks( $template_content );
+		$grouped_blocks = self::group_blocks( $parsed_blocks );
 
 		$single_product_template_blocks = array( 'woocommerce/breadcrumbs', 'woocommerce/product-gallery-image', 'woocommerce/product-add-to-cart', 'woocommerce/product-details', 'woocommerce/add-to-cart-form' );
 
-		if ( ! self::has_single_product_template_blocks( $parsed_block_without_template_parts, $single_product_template_blocks ) ) {
-			return $template_content;
-		}
+		$wrapped_blocks = array_map(
+			function( $blocks ) use ( $single_product_template_blocks ) {
+				$has_single_product_template_blocks = false;
```

Comment:
Can this line be removed? In L385 we are setting `$has_single_product_template_blocks` again.